### PR TITLE
MES-6880 Test Report Cat B Specific Components

### PR DESCRIPTION
**Pull request by tomcrawleyy**
## Description
Introduces the cat b specific components for the test report page
## Checklist

- [x] PR title includes the JIRA ticket number
- [x] Branch is rebased against the latest develop
- [x] Code has been tested manually
- [x] PR link added to JIRA ticket
- [ ] One review from each scrum team
- [x] Squashed commit contains the JIRA ticket number

### CODE REVIEW

File: src/app/pages/test-report/cat-b/components/manoeuvres-popover/manoeuvres-popover.html [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+          <manoeuvre-competency 
+            class="second" 
+            [manoeuvre]="manoeuvreTypes.reverseParkRoad"
+            [competency]="competencies.observationFault"
+            [attr.id]="getId(manoeuvreTypes.reverseParkRoad, competencies.observationFault)"
+          ></manoeuvre-competency>
+        </ion-col>
+      </ion-row>
+    </ion-grid>
+  </ng-container>
```
Comment (andrewsetterfield) [inline src/app/pages/test-report/cat-b/components/manoeuvres-popover/manoeuvres-popover.html]: git error, end of line.

File: src/app/pages/test-report/cat-b/components/manoeuvres-popover/manoeuvres.popover.scss [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+      margin-left: 68px;
+      border: 0 none var(--mes-white);
+    }
+    .manoeuvre-competencies-container {
+      margin-top: 40px;
+      .manoeuvre-competency-button {
+        margin-bottom: 10px;
+      }
+    }
+
```
Comment (andrewsetterfield) [inline src/app/pages/test-report/cat-b/components/manoeuvres-popover/manoeuvres.popover.scss]: git error, no EOF

File: src/app/pages/test-report/cat-b/components/manoeuvres/__tests__/manoeuvres.spec.ts [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+import { AppModule } from '@app/app.module';
+import { DangerousFaultBadgeComponent } from '@components/common/dangerous-fault-badge/dangerous-fault-badge';
+import { DrivingFaultsBadgeComponent } from '@components/common/driving-faults-badge/driving-faults-badge';
+import { SeriousFaultBadgeComponent } from '@components/common/serious-fault-badge/serious-fault-badge';
+import { TickIndicatorComponent } from '@components/common/tick-indicator/tick-indicator';
+import { IonicModule } from '@ionic/angular';
+import { StoreModule } from '@ngrx/store';
+import { configureTestSuite } from 'ng-bullet';
+import { MockComponent } from 'ng-mocks';
+import { DateTimeProvider } from '../../../../../../providers/date-time/date-time';
```
Comment (andrewsetterfield) [inline src/app/pages/test-report/cat-b/components/manoeuvres/__tests__/manoeuvres.spec.ts]: can these imports be aliased

File: src/app/pages/test-report/cat-b/components/manoeuvres/__tests__/manoeuvres.spec.ts [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+                journalData: {},
+                communicationPreferences: {
+                  updatedEmail: '',
+                  communicationMethod: 'Not provided',
+                  conductedLanguage: 'Not provided',
+                },
+              },
+            },
+          }),
+          // @TODO introduce this once Cat B NGRX introduced
```
Comment (andrewsetterfield) [inline src/app/pages/test-report/cat-b/components/manoeuvres/__tests__/manoeuvres.spec.ts]: do we have a story number for this?

File: src/app/pages/test-report/components/competency-button/__tests__/competency-button.spec.ts [hunk 5]
```diff
@@ -0,0 +1,164 @@
+import { ComponentFixture, async, TestBed } from '@angular/core/testing';
+import { configureTestSuite } from 'ng-bullet';
+import { AppModule } from '@app/app.module';
+import { IonicModule } from '@ionic/angular';
+import { By } from '@angular/platform-browser';
+import { CompetencyButtonComponent } from '../competency-button';
```
Comment (andrewsetterfield) [inline src/app/pages/test-report/components/competency-button/__tests__/competency-button.spec.ts:8]: can we use the aliases

File: src/app/pages/test-report/components/competency-button/competency-button.scss [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+      .remove-mode.serious-mode & {
+        z-index: 160;
+      }
+    }
+    &.dangerous-fault {
+      .remove-mode.dangerous-mode & {
+        z-index: 160;
+      }
+    }
+
```
Comment (andrewsetterfield) [inline src/app/pages/test-report/components/competency-button/competency-button.scss]: git error, no eof

File: src/app/pages/test-report/cat-b/components/vehicle-check/vehicle-check.scss [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+              margin-left: -22px;
+            }
+          }
+        }
+      }
+    
+      .show-me-tell-me-label {
+        font-size: 15px !important;
+      }
+
```
Comment (andrewsetterfield) [inline src/app/pages/test-report/cat-b/components/vehicle-check/vehicle-check.scss]: git error, no eof.

File: src/app/pages/test-report/cat-b/components/manoeuvres/__tests__/manoeuvres.spec.ts (hunk 4 above)
Comment (tomcrawleyy) [reply src/app/pages/test-report/cat-b/components/manoeuvres/__tests__/manoeuvres.spec.ts]: nice spot, matts ngrx has gone in now, will amend

File: src/app/pages/test-report/cat-b/components/vehicle-check/vehicle-check.html [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+        <driving-faults-badge class="driving-faults" [count]="getDrivingFaultCount()"></driving-faults-badge>
+        <div class='show-me-tell-me-label'>Show me / Tell me</div>
+        <div class="fault-wrapper">
+          <serious-fault-badge [showBadge]="hasSeriousFault()"></serious-fault-badge>
+          <dangerous-fault-badge [showBadge]="hasDangerousFault()"></dangerous-fault-badge>
+        </div>
+      </competency-button>
+    </ion-col>
+  </ion-row>
+
```
Comment (andrewsetterfield) [inline src/app/pages/test-report/cat-b/components/vehicle-check/vehicle-check.html]: missed one. EOF

File: src/app/pages/test-report/cat-b/components/manoeuvres-popover/manoeuvres-popover.html [hunk 9]
```diff
@@ -0,0 +1,102 @@
+<ng-container *ngIf="{
+    manoeuvres: manoeuvres$ | async
+  } as asyncData">
+  <ion-grid class="popover-content">
+    <ion-row>
+      <ion-col col-36>
```
Comment (Cullima) [inline src/app/pages/test-report/cat-b/components/manoeuvres-popover/manoeuvres-popover.html]: think we need to be changing these to size="36" ?

File: src/app/pages/test-report/components/competency-button/__tests__/competency-button.spec.ts (hunk 5 above)
Comment (andrewsetterfield) [reply src/app/pages/test-report/components/competency-button/__tests__/competency-button.spec.ts:8]: still here

File: src/app/pages/test-report/cat-b/components/manoeuvres-popover/manoeuvres-popover.ts [hunk 10]
```diff
@@ -0,0 +1,77 @@
...
+  reverseRight: boolean;
+  reverseParkRoad: boolean;
+  reverseParkCarpark: boolean;
+  forwardPark: boolean;
+  reverseLeft?: boolean;
+}
+
+@Component({
+  selector: 'manoeuvres-popover',
+  templateUrl: 'manoeuvres-popover.html',
```
Comment (Cullima) [inline src/app/pages/test-report/cat-b/components/manoeuvres-popover/manoeuvres-popover.ts:25]: need to add `styleUrls` to ensure css is picked up

File: src/app/pages/test-report/cat-b/components/manoeuvres-popover/manoeuvres.popover.scss [hunk 11]
```diff
@@ -0,0 +1,34 @@
+.gds-radio-button {
```
Comment (Cullima) [inline src/app/pages/test-report/cat-b/components/manoeuvres-popover/manoeuvres.popover.scss]: previous files was wrapped in `manoeuvres-popover{}` need to keep this but swap for `:host{}` (see conf page)

File: src/app/pages/test-report/cat-b/components/manoeuvres/manoeuvres.scss [hunk 12]
```diff
@@ -0,0 +1,117 @@
+button {
```
Comment (Cullima) [inline src/app/pages/test-report/cat-b/components/manoeuvres/manoeuvres.scss]: again, need to wrap with `:host{}` to define scope

File: src/app/pages/test-report/cat-b/components/vehicle-check/vehicle-check.html [hunk 13]
```diff
@@ -0,0 +1,37 @@
+<ion-row>
+    <ion-col col-20 no-padding>
```
Comment (Cullima) [inline src/app/pages/test-report/cat-b/components/vehicle-check/vehicle-check.html]: change to `size="20"`. Think you need to use `class="ion-no-padding"` now too

File: src/app/pages/test-report/cat-b/components/vehicle-check/vehicle-check.scss [hunk 14]
```diff
@@ -0,0 +1,40 @@
+competency-button {
```
Comment (Cullima) [inline src/app/pages/test-report/cat-b/components/vehicle-check/vehicle-check.scss]: `:host{}`

File: src/app/pages/test-report/components/competency-button/competency-button.scss [hunk 15]
```diff
@@ -0,0 +1,22 @@
+div.competency-button {
```
Comment (Cullima) [inline src/app/pages/test-report/components/competency-button/competency-button.scss]: `:host{}`

File: src/app/pages/test-report/components/competency-button/competency-button.ts [hunk 16]
```diff
@@ -0,0 +1,75 @@
+import { Component, Input } from '@angular/core';
+
+@Component({
+  selector: 'competency-button',
+  templateUrl: './competency-button.html',
```
Comment (Cullima) [inline src/app/pages/test-report/components/competency-button/competency-button.ts:5]: `styleUrls`

**Review by Cullima — CHANGES_REQUESTED** [review]

**Review by Cullima — APPROVED** [review]

**Review by andrewsetterfield — APPROVED** [review]

**Pull request merged**